charts/sftpgo/templates/ingress-api.yaml

```diff
@@ -0,0 +1,61 @@
+{{- if .Values.ingress.api.enabled -}}
+{{- $fullName := include "sftpgo.fullname" . -}}
+{{- $svcPort := .Values.service.ports.http.port -}}
+{{- if and .Values.ingress.api.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.api.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.api.annotations "kubernetes.io/ingress.class" .Values.ingress.api.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}-api
```

Comment:
```suggestion
  name: {{ include "sftpgo.componentname" (list . "api") }}
```

Reply:
Or better define a name in `templates/_helpers.tpl`?

Reply:
Not sure what you mean. The component name template was defined exactly for this purpose (making sure that the object name cannot be longer than the allowed length).